src/iso15118/d20/config.cpp

```diff
@@ -47,8 +77,14 @@ SessionConfig::SessionConfig(EvseSetupConfig config) :
                      "limits. This can lead to session shutdowns.");
     }
 
-    dc_parameter_list = get_default_dc_parameter_list();
-    dc_bpt_parameter_list = get_default_dc_bpt_parameter_list();
+    if (supported_control_mobility_modes.empty()) {
+        logf_warning("No control modes were provided, set to scheduled mode");
+        supported_control_mobility_modes = {
+            {message_20::ControlMode::Scheduled, message_20::MobilityNeedsMode::ProvidedByEvcc}};
+    }
+
+    dc_parameter_list = get_default_dc_parameter_list(supported_control_mobility_modes);
+    dc_bpt_parameter_list = get_default_dc_bpt_parameter_list(supported_control_mobility_modes);
```

Comment:
Note, you could also overload the functions and have a single name like `get_default_dc_parameter_list`.

Reply:
In this case I can not overload the functions. The function would only have different return values and that's not enough for function overloading.
But i can move the return value to a parameter. That works.